src/claim/entry/entry.js

```diff
@@ -1,33 +1,57 @@
+// @flow
+import { getArrayBigIntFromBuffArray, bigIntToBuffer } from '../../sparse-merkle-tree/sparse-merkle-tree-utils';
+
 const utils = require('../../utils');
 const mimc7 = require('../../sparse-merkle-tree/mimc7');
-const helpers = require('../../sparse-merkle-tree/sparse-merkle-tree-utils');
+
+const entryElemsLen = 4;
 
 /**
  * Generic representation of claim elements
  * Entry element structure is as follows: |element 0|element 1|element 2|element 3|
  * Each element contains 253 useful bits enclosed on a 256 bits Buffer
  */
 export class Entry {
+  _elements: Array<Buffer>;
+
+  constructor(e0: Buffer, e1: Buffer, e2: Buffer, e3: Buffer) {
+    this._elements = [e0, e1, e2, e3];
+  }
+
   /**
    * Initialize claim elements with empty buffer
    */
-  constructor() {
-    this._elements = [Buffer.alloc(32), Buffer.alloc(32), Buffer.alloc(32), Buffer.alloc(32)];
+  static newEmpty(): Entry {
+    return new Entry(Buffer.alloc(32), Buffer.alloc(32), Buffer.alloc(32), Buffer.alloc(32));
+  }
+
+  /**
+   * String deserialization into entry element structure
+   * @param {String} Hexadecimal string representation of element claim structure
+   */
+  static newFromHex(entryHex: string): Entry {
+    const entryBuff = utils.hexToBytes(entryHex);
+    const elements: any = [null, null, null, null];
+    for (let i = 0; i < entryElemsLen; i++) {
+      // Slice buffer into 32 bytes to insert it into an specific element
+      elements[(entryElemsLen - 1) - i] = entryBuff.slice(entryBuff.length - (32 * (i + 1)), entryBuff.length - 32 * i);
+    }
+    return new Entry(elements[0], elements[1], elements[2], elements[3]);
   }
 
   /**
    * Retrieve elements structure
    * @returns {Array} - Elements structure
    */
-  get elements() {
+  get elements(): Array<Buffer> {
```

Comment:
- We could remove getter / setter functions if we agree to skip that pattern.
- It would imply to rename `this._elements` to `this.elements`

Reply:
See commit 3fa5f4d